src/mongo/tree/MongoCollectionTreeItem.ts

```diff
@@ -101,46 +102,41 @@ export class MongoCollectionTreeItem implements IAzureParentTreeItem {
 		return new MongoDocumentTreeItem(newDocument, this.collection);
 	}
 
-	//tslint:disable:cyclomatic-complexity
-	executeCommand(name: string, args?: string[]): Thenable<string> {
+	executeCommand(name: string, args?: string[]): Thenable<string> | null {
+		//const requiresOneArg = ['findOne', 'insertMany', 'insertOne', 'insert', 'deleteOne', 'deleteMany', 'remove'];
+		const parameters = args ? args.map(parseJSContent) : undefined;
 		try {
-			if (name === 'findOne') {
-				return reportProgress(this.findOne(args ? args.map(parseJSContent) : undefined), 'Finding');
-			}
-			if (name === 'drop') {
-				return reportProgress(this.drop(), 'Dropping collection');
-			}
-			if (name === 'insertMany') {
-				return reportProgress(this.insertMany(args ? args.map(parseJSContent) : undefined), 'Inserting documents');
-			}
-			else {
-				let argument;
-				if (args && args.length > 1) {
-					return undefined;
-				}
-				if (args) {
-					argument = args[0];
-				}
-				if (name === 'insert') {
-					return reportProgress(this.insert(argument ? parseJSContent(argument) : undefined), 'Inserting document');
-				}
-				if (name === 'insertOne') {
-					return reportProgress(this.insertOne(argument ? parseJSContent(argument) : undefined), 'Inserting document');
-				}
-				if (name === 'deleteOne') {
-					return reportProgress(this.deleteOne(argument ? parseJSContent(argument) : undefined), 'Deleting document');
-				}
-				if (name === 'deleteMany') {
-					return reportProgress(this.deleteMany(argument ? parseJSContent(argument) : undefined), 'Deleting documents');
+			// tslint:disable-next-line:no-any
+			type MongoFunction = (...args: any[]) => Thenable<string>;
+			let functions: { [functionName: string]: [MongoFunction, string, number, number, number] } = {
+				// format: command name (from the argument) : corresponding function call[0],
+				// text to show during the operation[1], min #args[2], #max args[3], #args this function handles[4]
```

Comment:
easier to read as:
```
0: corresponding function call
1: min #args
```
etc


Also explain what "#args this function handles" means, it's not very clear